docs/proposals/edi.md

```diff
@@ -0,0 +1,258 @@
+   string date?;
+|};
+
+type Items_Type record {|
+   string item?;
+   int quantity?;
+|};
+
+type SimpleOrder record {|
+   Header_Type header;
+   Items_Type[] items?;
+|};
+```
+
+- **Using compiler plugin at the build time**
+
+  The ballerina compiler plugin can be configured to generate EDI records during the build time. This support for ballerina is
+  still under development and once it is available, EDI records can be generated using that feature.
+
+## Parsing EDI files
+
+The below code reads the edi-sample1.edi into a json variable named "orderData" and then convert the orderData json to the
+generated record "SimpleOrder". Once EDI documents are mapped to the SimpleOrder record, any attribute in the EDI can be
+accessed using record's fields as shown in the example code below.
+
+```ballerina
+import ballerina/io;
+import chathurace/edi.core as edi;
+
+public function main() returns error? {
+    edi:EDIMapping mapping = check edi:readMappingFromFile("resources/edi-mapping1.json");
+
+    string ediText = check io:fileReadString("resources/edi-sample1.edi");
+    json orderData = check edi:readEDIAsJson(ediText, mapping);
+    io:println(orderData.toJsonString());
+
+    SimpleOrder order1 = check orderData.cloneWithType(SimpleOrder);
+    io:println(order1.header.date);
+}
+```
+
+"orderData" json variable value will be as follows (i.e. output of io:println(orderData.toJsonString())):
+```json
+{
+  "header": {
+    "orderId": "ORDER_1201",
+    "organization": "ABC_Store",
+    "date": "2008-01-01"
+  },
+  "items": [
+    {
+      "item": "A-250",
+      "quantity": 12
+    },
+    {
+      "item": "A-45",
+      "quantity": 100
+    },
+    {
+      "item": "D-10",
+      "quantity": 58
+    },
+    {
+      "item": "K-80",
+      "quantity": 250
+    },
+    {
+      "item": "T-46",
+      "quantity": 28
+    }
+  ]
+}
+```
+
+A sample Ballerina project which uses the EDI library is given
+in [here](https://github.com/chathurace/ballerina-edi/tree/main/samples/simpleEDI)
+
+Also refer to [resources](https://github.com/chathurace/ballerina-edi/tree/main/edi/resources) section for example
+mapping files and edi samples.
+
+
+## Creating an EDI library
+
+Usually, organizations have to work with many EDI formats, and integration developers need to have a convenient way to
+work on EDI data with minimum effort. EDI libraries facilitate this by allowing organizations to pack all EDI processing
+code for to their EDI collections into an importable library. Therefore, integration developers can simply import those
+libraries and convert EDI messages into Ballerina records in a single line of code.
+
+Below command can be used to generate EDI libraries:
+
+```
+bal libgen <org name> <library name> <EDI mappings folder> <output folder>
+```
+
+The Ballerina library project will be generated in the output folder. This library can be built and published by issuing "
+bal pack" and "bal push" commands from the output folder.
```

Comment:
once we have the support for `generated` directories, we should be able to generate the same library under the `generated` directory under the same package and, import it in user's code (without having to publish to central)